private/private.go

```diff
@@ -58,42 +58,31 @@ func FromEnvironmentOrNil(name string) PrivateTransactionManager {
 }
 
 func MustNewPrivateTxManager(cfgPath string) PrivateTransactionManager {
-	info, err := os.Lstat(cfgPath)
+	cfg, err := engine.LoadConfig(cfgPath)
 	if err != nil {
 		panic(fmt.Sprintf("unable to read %s due to %s", cfgPath, err))
 	}
-	// We accept either the socket or a configuration file that points to
-	// a socket.
-	socketPath := cfgPath
-	isSocket := info.Mode()&os.ModeSocket != 0
-	if !isSocket {
-		cfg, err := engine.LoadConfig(cfgPath)
-		if err != nil {
-			panic(fmt.Sprintf("unable to load configuration file for private transaction manager from %s due to %s", cfgPath, err))
-		}
-		socketPath = filepath.Join(cfg.WorkDir, cfg.Socket)
-	}
 
 	client := &engine.Client{
 		HttpClient: &http.Client{
-			Transport: unixTransport(socketPath),
+			Transport: unixTransport(cfg),
 		},
 		BaseURL: "http+unix://c",
```

Comment:
You could set `http.Client.Timeout` here

Reply:
Ok, that could be useful. The user would need to modify `ClientTimeout` and the `ResponseHeaderTimeout` in conjunction though. That gives more flexibility (albeit with more potential for getting it wrong).